transport/x/grpc/inbound.go

```diff
@@ -134,45 +135,42 @@ func (i *Inbound) getServiceDescs() ([]*grpc.ServiceDesc, error) {
 	if len(procedures) == 0 {
 		return nil, errRouterHasNoProcedures
 	}
-	grpcServiceNameToServiceDesc := make(map[string]*grpc.ServiceDesc)
+
+	serviceNameToMethodNames := make(map[string]map[string]bool)
 	for _, procedure := range procedures {
-		serviceName, methodDesc, err := i.getServiceNameAndMethodDesc(procedure)
+		serviceName, methodName, err := procedureNameToServiceNameMethodName(procedure.Name)
 		if err != nil {
 			return nil, err
 		}
-		serviceDesc, ok := grpcServiceNameToServiceDesc[serviceName]
+		methodNames, ok := serviceNameToMethodNames[serviceName]
 		if !ok {
-			serviceDesc = &grpc.ServiceDesc{
-				ServiceName: serviceName,
-				HandlerType: (*noopGrpcInterface)(nil),
-			}
-			grpcServiceNameToServiceDesc[serviceName] = serviceDesc
+			methodNames = make(map[string]bool)
+			serviceNameToMethodNames[serviceName] = methodNames
 		}
-		serviceDesc.Methods = append(serviceDesc.Methods, methodDesc)
+		methodNames[methodName] = true
 	}
-	serviceDescs := make([]*grpc.ServiceDesc, 0, len(grpcServiceNameToServiceDesc))
-	for _, serviceDesc := range grpcServiceNameToServiceDesc {
+
+	serviceDescs := make([]*grpc.ServiceDesc, 0, len(serviceNameToMethodNames))
```

Comment:
aside - looks like this big block might also be more neatly tucked away into its own func